src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -6047,14 +6047,14 @@ void MacroAssembler::generate_fill(BasicType t, bool aligned,
 
           vpbroadcastd(xtmp, xtmp, Assembler::AVX_512bit);
 
-          subl(count, 16 << shift);
+          subptr(count, 16 << shift);
```

Comment:
At line 6045 also the cmpl should change to cmpptr:
cmpl(count, VM_Version::avx3_threshold());

Reply:
Will do.